doc/design/cleanup.md

```diff
@@ -0,0 +1,31 @@
+# Cleanup design
+
+Cleanup: Operations needed to do when the lvm cluster CRD is removed.
```

Comment:
This design will be heavily influenced by discussion about the design discussion for whether we have a single LVMCluster CR, or multiple LVMVolumeGroup CRs. This current suggestion assumes the LVMCluster design. With the LVMVolumeGroup design, I see deletion steps to be the following...

When a LVMVolumeGroup is deleted:
1. There is a finalizer on the LVMVolumeGroup that prevents deletion while the cleanup is being processed.
2. The storage class for the LVMVolumeGroup is deleted, so no new PVs can be requested.
3. The operator updates the configuration for each node to signal the removal of the deleted LVMVolumeGroup
4. Each node cleans up:  
   a. If there are any LVs in the VG, block deletion
   b. If there are no LVs in the VG, delete the VG
   c. Depending on the deletion policy on the LVMVolumeGroup (which we can discuss in the future), either wipe the disk(s), or do nothing.